include/extractor/io.hpp

```diff
@@ -96,6 +99,51 @@ inline void write(const boost::filesystem::path &path, const SegmentDataContaine
     writer.WriteFrom(segment_data.rev_durations);
     writer.WriteFrom(segment_data.datasources);
 }
+
+// read/write for conditional turn restrictions file
+inline void read(const boost::filesystem::path &path, std::vector<InputRestrictionContainer> &restrictions)
+{
+    const auto fingerprint = storage::io::FileReader::VerifyFingerprint;
+    storage::io::FileReader reader{path, fingerprint};
+
+    auto num_indices = reader.ReadElementCount32();
+    restrictions.resize(num_indices);
+    while (num_indices > 0)
+    {
+        InputRestrictionContainer res;
+        bool is_only;
+        reader.ReadInto(res.restriction.via);
+        reader.ReadInto(res.restriction.from);
+        reader.ReadInto(res.restriction.to);
+        reader.ReadInto(is_only);
+        for (auto &c : res.restriction.condition)
+        {
+            reader.ReadInto(c.modifier);
+            reader.DeserializeVector(c.times);
+            reader.DeserializeVector(c.weekdays);
+            reader.DeserializeVector(c.monthdays);
+        }
+        res.restriction.flags.is_only = is_only;
+        restrictions.push_back(res);
```

Comment:
Above the code resizes the `restrictions` vector to size `num_indices`, but here you `push_back` into it.

I think what we want is either
- To `.reserve(num_indices)` above (tells the vector we expect roughly `num_incices` elements) and then `push_back` here or
- To `.resize(num_indices)` above and assign to `restrictions[i]` here

Think of `reserve` as a hint for vector to allocate memory more efficiently. In contrast, `resize(n)` adds `n` elements to the vector (default constructed).